### ISSUE TITLE
Multi-select dropdown "Selected" state not announced by screen reader

### ISSUE THREAD

**Issue by austinvanderkarr-code** [jpmorganchase/salt-ds#5405]
### Latest version

- [x] I have tested the latest version

### Description

When a box is checked in the multi-select dropdown, the screen reader should announce "Selected".

Currently, the screen reader announces "not selected" when an unchecked item is in focus, but does not announce any state when a checked item is in focus. Both states should be announced by the screen reader.

Note: when a box is checked/unchecked, the screen reader correctly announces the changed state ("selected"/"unselected"), so the behavior there is already fine.

### Steps to reproduce

1.  Check some boxes in the multi-select dropdown
2. Enable screen reader
3. Navigate through items in multi-select dropdown

### Expected behavior

When an item with a checked box is in focus, the screen reader includes "selected" in its announcement.

### Package name(s)

Core (@salt-ds/core)

### Browser

- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge

### Operating system

- [ ] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
This behaviour is dependent on the screenreader. The dropdown has the correct attributes applied, so although it may be unexpected I believe the behaviour is correct.

What are your thoughts @jake-costa?

I'm going to close this issue for now but will re-open if a different stance is decided.
